python/setup.py

```diff
@@ -560,13 +560,18 @@ def build(build_python, build_java, build_cpp):
             FutureWarning,
         )
 
-    if not is_automated_build:
-        bazel_precmd_flags = []
     if is_automated_build:
-        root_dir = os.path.join(
-            os.path.abspath(os.environ["SRC_DIR"]), "..", "bazel-root"
-        )
-        out_dir = os.path.join(os.path.abspath(os.environ["SRC_DIR"]), "..", "b-o")
+        if is_native_windows_or_msys():
```

Comment:
qq: is everything under this if condition applies only to conda-forge? If so perhaps put them all into a function so that the normal conda distribution won't mess it up.

Reply:
I don't have access to the normal conda distribution code, so I don't know. This code came in from #17575 by @krfricke for the conda-forge feedstock. I would think everyone setting `AUTOMATED_BUILD` on windows will need this to shorten the path enough so that bazel will succeed.

Reply:
Sound good, I think addressing Lonnie's comments and we are good to go